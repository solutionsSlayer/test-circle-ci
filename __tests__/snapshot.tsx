/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import Home from "@/pages/home";

jest.mock('next/head', () => {
  return {
    __esModule: true,
    default: ({ children }: { children: React.ReactNode }) => <>{children}</>,
  };
});

it("renders homepage unchanged", async () => {
  const { container } = render(<Home />);
  await new Promise(resolve => setTimeout(resolve, 0)); // Allow async operations to complete
  expect(container).toMatchSnapshot();
});
