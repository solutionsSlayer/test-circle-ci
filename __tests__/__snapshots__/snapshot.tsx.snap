// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <h1>
        Calculator
      </h1>
      <div
        class="flex"
      >
        <input
          placeholder="Number A"
          type="number"
          value=""
        />
        <select>
          <option
            disabled=""
            value=""
          >
            Select operation
          </option>
          <option
            value="+"
          >
            +
          </option>
          <option
            value="-"
          >
            -
          </option>
          <option
            value="*"
          >
            *
          </option>
          <option
            value="/"
          >
            /
          </option>
        </select>
        <input
          placeholder="Number B"
          type="number"
          value=""
        />
      </div>
      <button>
        Calculate
      </button>
      <p>
        Result: 
      </p>
      <h3>
        History:
      </h3>
      <ul />
    </div>
  </div>
</div>
`;
